### PR TITLE
📚 Add kdoc for Square.* and Testing., ...

### PR DESCRIPTION
**Pull request by jmfayard**

### CODE REVIEW

**Comment by jmfayard** [comment]
do not merge yet, will continue later

**Comment by jmfayard** [comment]
ready for review

File: plugins/dependencies/src/main/kotlin/dependencies/Kotlin.kt
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/Kotlin.kt:21]: These dangling links are not clickable in the IDE. Let's avoid that and always have text setting expectations on what the link is about.

File: plugins/dependencies/src/main/kotlin/dependencies/CashApp.kt
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/CashApp.kt:12]: The style here is inconsistent with the existing dependency notations KDoc that can be found in the `Square` object.

Having a consistent style makes a better impression on users and values the effort in making the whole project right.

Happy to improve the style, but we need to discuss it and agree on a style to stick with.
I'll complete a contributing guide covering this as we might welcome further contributions there from the community, and we might forget about it ourselves.

I'm not repeating this comment to all the other KDocs in this PR, this one stands for them all.

File: plugins/dependencies/src/main/kotlin/dependencies/KotlinX.kt
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/KotlinX.kt:68]: Dangling links

File: plugins/dependencies/src/main/kotlin/dependencies/Orchid.kt
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/Orchid.kt:22]: Dangling link

File: plugins/dependencies/src/main/kotlin/dependencies/Orchid.kt
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/Orchid.kt:11]: Dangling link

File: plugins/dependencies/src/main/kotlin/dependencies/Square.kt
Comment (LouisCAD) [inline plugins/dependencies/src/main/kotlin/dependencies/Square.kt:25]: We don't control how outdated a third-party article can become, yet, we are helping our users update (diverging further from these possibly outdated articles at a quicker rate).

Also, we might never find the best article on a library, so let's stick to the official and blog posts from authors and major contributors, and for the rest, or when official content is lacking, let the users decide what to look for as a reference based on the metrics they value like time of publication, programming language (Kotlin vs Java), platform (Android, JVM, other…), search engines and critical sense do a great job for that.

**Review by LouisCAD — CHANGES_REQUESTED** [review]
I've already started correcting the style to be consistent, and addressing other comments I made.

I'll merge this once I'm done with it (will happen before Monday).

**Review by LouisCAD — APPROVED** [review]

**Comment by LouisCAD** [comment]
@jmfayard Feel free to review the changes I just pushed and merge once the CI is done running the checks.

**Pull request merged**
